From a69e53b3c8adf726d106d7666702c6d589b41369 Mon Sep 17 00:00:00 2001
From: Liam Girdwood <lrg@ti.com>
Date: Thu, 3 Feb 2011 18:15:17 +0000
Subject: [PATCH 42/54] ASoC: OMAP4 - set constraints for buffer size

ABE needs a buffer size in steps of 24 * 4 data bits. HDMI requires
steps of 32 * 4. Ensure buffer size satisfies both constraints.

Signed-off-by: Liam Girdwood <lrg@ti.com>
---
 sound/soc/omap/omap-pcm.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/sound/soc/omap/omap-pcm.c b/sound/soc/omap/omap-pcm.c
index b095727..d429791 100644
--- a/sound/soc/omap/omap-pcm.c
+++ b/sound/soc/omap/omap-pcm.c
@@ -279,6 +279,15 @@ static int omap_pcm_open(struct snd_pcm_substream *substream)
 					    SNDRV_PCM_HW_PARAM_PERIODS);
 	if (ret < 0)
 		goto out;
+	if (cpu_is_omap44xx()) {
+		/* ABE needs a step of 24 * 4 data bits, and HDMI 32 * 4
+		 * Ensure buffer size satisfies both constraints.
+		 */
+		ret = snd_pcm_hw_constraint_step(runtime, 0,
+					 SNDRV_PCM_HW_PARAM_BUFFER_BYTES, 384);
+		if (ret < 0)
+			goto out;
+	}
 
 	prtd = kzalloc(sizeof(*prtd), GFP_KERNEL);
 	if (prtd == NULL) {
-- 
1.7.4.1

